pkg/apis/storage/validation/validation_test.go

```diff
@@ -1772,3 +1826,94 @@ func TestCSIDriverValidation(t *testing.T) {
 		}
 	}
 }
+
+func TestCSIDriverValidationUpdate(t *testing.T) {
+	driverName := "test-driver"
+	longName := "my-a-b-c-d-c-f-g-h-i-j-k-l-m-n-o-p-q-r-s-t-u-v-w-x-y-z-ABCDEFGHIJKLMNOPQRSTUVWXYZ-driver"
+	invalidName := "-invalid-@#$%^&*()-"
+	attachRequired := true
+	attachNotRequired := false
+	podInfoOnMount := true
+	notPodInfoOnMount := false
+	old := storage.CSIDriver{
+		ObjectMeta: metav1.ObjectMeta{Name: driverName},
+		Spec: storage.CSIDriverSpec{
+			AttachRequired: &attachNotRequired,
+			PodInfoOnMount: &notPodInfoOnMount,
+			VolumeLifecycleModes: []storage.VolumeLifecycleMode{
+				storage.VolumeLifecycleEphemeral,
+				storage.VolumeLifecyclePersistent,
+			},
+		},
+	}
+
+	// Currently there is only one success case: exactly the same
+	// as the existing object.
+	successCases := []storage.CSIDriver{old}
+	for _, csiDriver := range successCases {
+		if errs := ValidateCSIDriverUpdate(&csiDriver, &old); len(errs) != 0 {
+			t.Errorf("expected success for %+v: %v", csiDriver, errs)
+		}
+	}
+
+	errorCases := []storage.CSIDriver{
+		{
```

Comment:
These first few error test cases are actually testing two changes:
* Changing name or AttachedRequired, PodInfoOnMount fields
* Dropping VolumeLifecycleModes

It would be better if the error case only changed one thing at a time so that one failure reason doesn't mask the other.